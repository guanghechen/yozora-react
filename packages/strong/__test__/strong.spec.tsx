import React from 'react'
import { mount, render } from 'enzyme'
import Strong from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render a simple content', () => {
    const text = 'Hello, world!'
    const wrapper = render(
      <Strong>
        <span>
          <Strong>{ text }</Strong>
        </span>
      </Strong>
    )
    expect(wrapper.text()).toEqual(text)
  })

  it('render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-text'
    const wrapper = render(
      <Strong className={ className }>
        <span>{ text }</span>
      </Strong>
    )
    expect(wrapper.hasClass(className)).toEqual(true)
    expect(wrapper.text()).toEqual(text)
  })

  it('children is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        mount(
          <Strong>
            { value }
          </Strong>
        )
      }).toThrow(/Failed prop type: The prop `children` is marked as required/)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLSpanElement>()
    const wrapper = mount(
      <Strong ref={ ref } data-value="waw">
        1
      </Strong>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = mount(
      <Strong style={{ color: 'orange', fontSize: '16px' }}>
        some text1
        <span>some text2</span>
      </Strong>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
