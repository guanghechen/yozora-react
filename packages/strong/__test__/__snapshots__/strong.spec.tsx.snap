// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot custom 1`] = `
<strong
  class="custom-class"
  data-name="yozora-strong"
  style="color:orange"
>
  some text1
  <span>
    some text2
  </span>
</strong>
`;

exports[`snapshot default 1`] = `
<strong
  class="yozora-strong"
>
  some text1
  <span>
    some text2
  </span>
</strong>
`;
